src/storage-blob-preview/azext_storage_blob_preview/operations/blob.py

```diff
@@ -574,6 +574,22 @@ def snapshot_blob(client, metadata=None, **kwargs):
     return client.get_blob_properties()
 
 
+# pylint: disable=protected-access
+def _adjust_block_blob_size(client, blob_type, length):
+    if not blob_type or blob_type != 'block':
+        return
+    # increase the block size to 100MB when the block list will contain more than
+    # 50,000 blocks(each block 4MB)
+    if length > 50000 * 4 * 1024 * 1024:
```

Comment:
should also make sure length <= 50000 * 100 * 1024 * 1024

Reply:
For those larger than 50000 * 100 * 1024 * 1024, the best thing we can do is making the config biggest value. So I didn't add length <= 50000 * 100 * 1024 * 1024 here.